src/test/java/BaseTest.java

```diff
@@ -1,11 +1,132 @@
+        arrObj = getExcelData("./src/test/resources/test.xlsx", "Sheet1");
+        return arrObj;
+    }
+
+    public String[][] getExcelData(String fileName, String sheetName){
+        String [][] data = null;
+        try{
+            FileInputStream fileInputStream = new FileInputStream(fileName);
+            XSSFWorkbook wb = new XSSFWorkbook(fileInputStream);
+            XSSFSheet sheet = wb.getSheet(sheetName);
+            XSSFRow row = sheet.getRow(0);
+
+            int numOfRows = sheet.getPhysicalNumberOfRows();
+            int numOfColumns = row.getLastCellNum();
+
+            XSSFCell cell;
+            data = new String[numOfRows-1][numOfColumns];
+
+            for(int i=1; i<numOfRows; i++) {
+                for(int j=0; j<numOfColumns; j++){
+                    row = sheet.getRow(i);
+                    cell = row.getCell(j);
+                    data [i-1][j] = cell.getStringCellValue();
+                }
+            }
+        }  catch (Exception e) {
+            System.out.println("Something went wrong." +e);
+        }
+        return data;
+    }
+    //DataProvider end
+
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
+ @BeforeMethod
+ @Parameters({"BaseURL"})
+    public void launchBrowser(String BaseURL){
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--remote-allow-origins=*");
+
+        driver = new ChromeDriver(options);
+        //driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10)); - Implicitly Wait
+     //Explicitly Wait:
+        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        action = new Actions(driver);
+        driver.manage().window().maximize();
+        url = BaseURL;
+        navigateToLoginPage();
+
+    }
+@AfterMethod
+    public void closeBrowser(){
+        driver.quit();
+    }
+
+    public void navigateToLoginPage(){
+        driver.get(url);
+    }
+
+    public void provideEmail(String email){
+        WebElement emailField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("input[type='email']")));
+        //WebElement emailField = driver.findElement(By.cssSelector("input[type='email']"));
+        emailField.clear();
+        emailField.sendKeys(email);
+    }
+
+    public void providePassword(String password){
+        WebElement passwordField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("input[type='password']")));
+        //WebElement passwordField = driver.findElement(By.cssSelector("input[type='password']"));
+        passwordField.clear();
+        passwordField.sendKeys(password);
+    }
+
+    public void clickSubmit(){
+        WebElement submit = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("button[type='submit']")));
+        //WebElement submit = driver.findElement(By.cssSelector("button[type='submit']"));
+        submit.click();
+    }
```

Comment:
Base Page \ Base Class — Implements the necessary methods for work with webdriver.

We will store the web elements present on each page in the corresponding java class.

The methods `provideEmail , providePassword, clickSubmit must be stored in the `class LoginPage` because the avatar is visible on the web site -home page